app/src/main/java/com/fedorov/alex/whattheweatheristoday/MainActivity.java

```diff
@@ -1,22 +1,183 @@
 package com.fedorov.alex.whattheweatheristoday;
 
+import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.View;
 import android.widget.ImageView;
 import android.widget.TextView;
 
+import static android.content.Intent.ACTION_VIEW;
+
+
 public class MainActivity extends AppCompatActivity {
 
+
+    public final static String CITY = "CITY";
+
+    private static final String TAG = "myLogs";
+
+    private final static DataCache dataCache = DataCache.getDataCache();
+    public static final int REQUEST_CODE = 1;
+
+    TextView temperature;
+    ImageView weatherImage;
+    TextView cityView;
+
+    String currentCity;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        cityView = findViewById(R.id.city);
+        temperature = findViewById(R.id.temperature);
+        weatherImage = findViewById(R.id.weatherImage);
+
+        setButtonsOnClickListener();
+
+        setPreviousInstanceState();
     }
 
-  
 
+    private void setPreviousInstanceState() {
+        String prevCity = dataCache.getCity();
+        String prevTemp = dataCache.getTemp();
+        int prevImageRes = dataCache.getImageRes();
+
+        if (prevCity != null) {
+            cityView.setText(prevCity);
+        }
+
+        if (prevTemp != null) {
+            temperature.setText(prevTemp);
+        }
+
+        if (prevImageRes != -1) {
+            weatherImage.setImageResource(prevImageRes);
+        }
+    }
+
+    private void setButtonsOnClickListener() {
+        View.OnClickListener onClickListener = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                pushTheButton(view);
+            }
+        };
+
+        findViewById(R.id.sun_small).setOnClickListener(onClickListener);
+        findViewById(R.id.cloud_small).setOnClickListener(onClickListener);
+        findViewById(R.id.rain_small).setOnClickListener(onClickListener);
+        findViewById(R.id.snow_small).setOnClickListener(onClickListener);
```

Comment:
Вы можете прямо так вот сразу сказать, какой кнопке какой case в switch относится?

можно как вариант сделать несколько методов, и их вызывать в onClick

 ```suggestion
        findViewById(R.id.snow_small).setOnClickListener(new View.OnCLickListener(){ public void onClick(View v) { setSnow_small(); } });
```

И дальше метод setShow_small()